airbyte-integrations/connectors/source-notion/source_notion/source.py

```diff
@@ -33,10 +34,21 @@ def get_access_token(self):
 
 
 class SourceNotion(AbstractSource):
-    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+    @staticmethod
+    def set_start_date(config: Mapping[str, Any]):
```

Comment:
Would this have been safer to implement in `NotionStream.__init__`? I fear that every time we are instantiating a stream now, we need to think about calling `set_start_date` before. If we forget, the behavior might not be the one we expect

Reply:
That's a valid concern! To guarantee the `start_date` is initiated properly in each stream I've moved the logic to `NotionStream.__init__` as suggested. If no `start_date` is found in config, the stream will set it using the two year logic and update the value in config for the next streams to use as well.